### ISSUE TITLE
NXDOMAIN returned by the authoritative server for the empty non-terminals

### ISSUE THREAD

**Issue by SivaKesava1** [samboy/MaraDNS#86]
Hi again,

I found one more issue by using the following small zone file as input to the authoritative server. I hope this is a valid zone file to be an input to the authoritative server.

|  	|  	|  	|  	|
|-	|-	|-	|-	|
| example.com. 	| +60480 	| SOA 	| ns1.outside.com. root.example.com. 3 604800 86400 2419200 604800 ~ 	|
| example.com. 	| +60480 	| NS 	| ns1.outside.com. ~ 	|
| b.g.example.com. 	| +60480 	| NS 	| n.b.g.example.com. ~ 	|
| n.b.g.example.com. 	| +60480 	| A 	| 1.1.1.1 ~ 	|

For the query _<g.example.com. , A>_, the MaraDNS server returns the following response:
```
          "rcode NXDOMAIN",
          "flags QR AA",
          ";QUESTION",
          "g.example.com. IN A",
          ";ANSWER",
          ";AUTHORITY",
          "example.com. 604800 IN SOA ns1.outside.edu. root.campus.edu. 3 604800 86400 2419200 604800",
          ";ADDITIONAL",
```

Here, `g.example.com` does not exist, but a child exists, so the response code should be `NOERROR` as per the latest [RFC 8020 section 3.1](https://tools.ietf.org/html/rfc8020#section-3.1).

**Comment by samboy** [comment]
Thank you for the detailed and informative report.

Any RFC after 2010 (RFC8020 is a 2016 RFC) is a wishlist item. 

I’m closing this bug, not because I will never fix this, but because this is a wishlist item, not a bug report.

I have stopped trying to keep playing the “RFCs breaking previously RFC compliant code” game.

**Comment by samboy** [comment]
I will, of course, be open to pull requests which implement this in the way which pedantically follows the post-2010 RFCs.

**Comment by samboy** [comment]
To implement this one will mean that one will have to add a bunch of “don’t NXDOMAIN this” records to the hash. When I implemented this part of MaraDNS, back in 2001, there was no RFC requirement to change whether or not one gets a NXDOMAIN if child records exist. So, I used a hash as the data structure for storing records. Simple, fast. 

These things in the RFC like this assume a particular data structure being used to store records which MaraDNS does not use. There _are_ ways to work around this, but it would be quite a bit of trouble. I will change this to “WONTFIX”.
